torchvision/models/_api.py

```diff
@@ -76,6 +75,7 @@ def __getattr__(self, name):
         return super().__getattr__(name)
 
     def __deepcopy__(self, memodict=None):
+        # TODO: this isn't a real deep-copy? https://github.com/pytorch/vision/pull/6883#discussion_r1011372296
```

Comment:
@pmeier I still have concerns about https://github.com/pytorch/vision/pull/6883#discussion_r1011372296

We're pretending that those `Weights` members support deep-copy, but they really don't and this could lead to unexpected behaviour.

Is there a way to be more conservative here while still addressing the original issues of https://github.com/pytorch/vision/issues/6871?

Reply:
Discussion is happening in #7107.